includes/reader-activation/class-reader-activation.php

```diff
@@ -2001,10 +2001,16 @@ public static function process_auth_form() {
 		self::set_auth_intention_cookie( $email );
 
 		$user = \get_user_by( 'email', $email );
-		if ( ( ! $user && 'register' !== $action ) || ( $user && ! self::is_user_reader( $user ) ) ) {
+		if ( ! $user && 'register' !== $action ) {
 			return self::send_auth_form_response( new \WP_Error( 'unauthorized', wp_kses_post( __( 'Account not found. <a data-set-action="register" href="#register_modal">Create an account</a> instead?', 'newspack-plugin' ) ) ) );
 		}
 
+		if ( ! self::is_user_reader( $user ) ) {
```

Comment:
This is not taking into account the account registration flow and will throw a fatal error if I attempt to register via the auth modal. Needs a `! $user` check.

Reply:
Fix at #3944